src/GoTrueClient.ts

```diff
@@ -2607,7 +2610,9 @@ export default class GoTrueClient {
 
     // try fetching from cache
     jwk = this.jwks.keys.find((key) => key.kid === kid)
-    if (jwk) {
+
+    // jwk exists and jwks isn't stale
+    if (jwk && Date.now() - JWKS_TTL < this.jwks_cached_at) {
```

Comment:
Any chance we can use better date logic here? `Date` objects are comparable also.

Reply:
hmm what did you have in mind? we pretty much just do an inequality check in most places - i've rearranged the vars to make it abit clearer